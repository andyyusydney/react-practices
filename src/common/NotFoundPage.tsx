import * as React from 'react'

export default () => <h1>Oops! Page is Not Found!</h1>