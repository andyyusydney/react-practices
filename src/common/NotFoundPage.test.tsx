// src/__test__/notFound.spec.tsx
import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import NotFoundPage from "./NotFoundPage";

let documentBody: RenderResult;
describe("<NotFound />", () => {
  beforeEach(() => {
    documentBody = render(<NotFoundPage />);
  });

  it("shows not found message", () => {
    expect(
      documentBody.getByText("Oops! Page is Not Found!")
    ).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
