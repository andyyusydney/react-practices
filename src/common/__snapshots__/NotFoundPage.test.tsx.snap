// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> matches snapshot 1`] = `
<body>
  <div>
    <h1>
      Oops! Page is Not Found!
    </h1>
  </div>
</body>
`;
